src/extension/extensionServer.ts

```diff
@@ -231,51 +231,55 @@ export class ExtensionServer implements vscode.Disposable {
-                        generator.step("startPackager");
-                        return mobilePlatform.startPackager();
-                    })
-                    .then(() => {
-                        // We've seen that if we don't prewarm the bundle cache, the app fails on the first attempt to connect to the debugger logic
-                        // and the user needs to Reload JS manually. We prewarm it to prevent that issue
-                        generator.step("prewarmBundleCache");
-                        this.logger.info(localize("PrewarmingBundleCache", "Prewarming bundle cache. This may take a while ..."));
-                        return mobilePlatform.prewarmBundleCache();
-                    })
-                    .then(() => {
-                        generator.step("mobilePlatform.runApp").add("target", mobilePlatformOptions.target, false);
-                        this.logger.info(localize("BuildingAndRunningApplication", "Building and running application."));
-                        return mobilePlatform.runApp();
-                    })
-                    .then(() => {
-                        if (mobilePlatformOptions.isDirect) {
-                            generator.step("mobilePlatform.enableDirectDebuggingMode");
-                            if (request.arguments.platform === "android") {
-                                this.logger.info(localize("PrepareHermesDebugging", "Prepare Hermes debugging (experimental)"));
-                            }
-                            return mobilePlatform.disableJSDebuggingMode();
-                        }
-                        generator.step("mobilePlatform.enableJSDebuggingMode");
-                        this.logger.info(localize("EnableJSDebugging", "Enable JS Debugging"));
-                        return mobilePlatform.enableJSDebuggingMode();
-                    })
-                    .then(() => {
-                        resolve();
-                    })
-                    .catch(error => {
-                        generator.addError(error);
-                        this.logger.error(error);
-                        reject(error);
+            ReactNativeProjectHelper.getReactNativeVersionFromProjectPackage(mobilePlatformOptions.projectRoot)
+                .then(version => {
+                    TelemetryHelper.addReactNativeVersionToEventProperties(version, extProps);
+                    TelemetryHelper.generate("launch", extProps, (generator) => {
+                        generator.step("checkPlatformCompatibility");
+                        TargetPlatformHelper.checkTargetPlatformSupport(mobilePlatformOptions.platform);
+                        return mobilePlatform.beforeStartPackager()
+                            .then(() => {
+                                generator.step("getReactNativeVersion");
+                                return ReactNativeProjectHelper.getReactNativePackageVersionFromNodeModules(mobilePlatformOptions.workspaceRoot);
```

Comment:
Do we need this after that?
https://github.com/microsoft/vscode-react-native/pull/1147/files#diff-a0bb3c3c1809e7b5cf209c8263767d8fR234

Reply:
`getReactNativeVersionFromProjectPackage` function gets React Native version only from project's `package.json`. There may be a case when react-native package isn't installed, but `getReactNativeVersionFromProjectPackage` function doesn't check project's `node_modules`. So it's nessesary to verify if `react-native` package is installed and `getReactNativePackageVersionFromNodeModules` function allows to do it. But I'll try to find more effective solution rather to check React Native version twice.

Reply:
Fixed. I changed React Native version getting behavior. I also changed sending telemetry events.